lib/src/util/leak_detector_visitor.dart

```diff
@@ -147,15 +146,15 @@ abstract class LeakDetectorVisitor extends SimpleAstVisitor {
 
   @override
   void visitFieldDeclaration(FieldDeclaration node) {
-    ClassDeclaration classDecl = node.getAncestor((a) => a is ClassDeclaration);
+    CompilationUnit unit = node.getAncestor((a) => a is CompilationUnit);
     node.fields.variables.forEach(_buildVariableReporter(
-        classDecl, _fieldPredicateBuilders, rule, predicates));
+        unit, _fieldPredicateBuilders, rule, predicates));
   }
 
   @override
   void visitVariableDeclarationStatement(VariableDeclarationStatement node) {
-    FunctionBody function = node.getAncestor((a) => a is FunctionBody);
+    CompilationUnit unit = node.getAncestor((a) => a is CompilationUnit);
```

Comment:
It isn't clear to me what use case this is covering. It seems like VariableDeclarationStatements should only be possible inside function bodies, and the variables cannot be referenced outside the function. What am I forgetting?

If there is a case, let's add a test that covers the case.